### PR TITLE
Add permissions.OperandHolder to decorators.permission_classes signature

### PR DESCRIPTION
**Pull request by kontsaki**
Hello, this will allow `permission_classes` to accept types like `BasePermission & BasePermission`.
i am not sure if i should import `_PermissionsClasses` as defined in the permissions stub and remove the preceding underscore, feedback is appreciated.

### CODE REVIEW

**Review by sobolevn — CHANGES_REQUESTED** [review]
Hi, thanks! Let's add for this!

**Comment by kontsaki** [comment]
oh yes, on it.

**Comment by kontsaki** [comment]
would a test like this suffice ?

File: tests/typecheck/test_decorators.yml [hunk 1]
```diff
@@ -24,3 +24,8 @@
         from rest_framework.decorators import api_view
         @api_view()  # E: Value of type variable "_View" of function cannot be "Callable[[Any], List[Any]]"
         def view_func2(request) -> list: ...
+
+-   case: permission_classes
+    main: |
+        from rest_framework.decorators import permission_classes
```
Comment (sobolevn) [inline tests/typecheck/test_decorators.yml:30]: We need to test that `&` form works. You can also add tests for other ways if
- they are missing
- you have the time for it 🙂

**Comment by kontsaki** [comment]
i can add more tests if you have something else in mind

**Review by sobolevn — APPROVED** [review]
Thanks!

**Pull request merged**